### PR TITLE
Update filter

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

File: amqp/src/AmqpClient.cpp [hunk 1]
```diff
@@ -99,17 +99,15 @@ void AmqpClient::on_connection_close(proton::connection&)
...
-        logInfo("Execute communication error callback");
-        // Execute callback treatment
-        m_pool->offload(std::bind(m_errorListener));
-    }
+    // reconnection option parameters. The idea is to try to re-open directly the connection.
+    // Note: The container must have the option auto_stop deactivated to not stop completely the container.
+    // Without that, the communication restoration will be more difficult.
+
+    m_connection = connection;
+    connection.work_queue().add([=]() { m_connection.open(); });
```
Comment (perrettecl) [inline amqp/src/AmqpClient.cpp:110]: Is the HOTFIX comment still valid ?

File: amqp/tests/MessageBusAmqpTest.cpp [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #include <catch2/catch.hpp>
 #include <fty/messagebus2/Message.h>
 #include <fty/messagebus2/MessageBusStatus.h>
+#include "../src/AmqpClient.h"
```
Comment (perrettecl) [inline amqp/tests/MessageBusAmqpTest.cpp:23]: Do we need this include ?

**Review by perrettecl — APPROVED** [review]

File: amqp/tests/MessageBusAmqpTest.cpp (hunk 2 above)
Comment (ericclappier-eaton) [reply amqp/tests/MessageBusAmqpTest.cpp:23]: Yes for small filter function testing.

File: amqp/src/AmqpClient.cpp (hunk 1 above)
Comment (ericclappier-eaton) [reply amqp/src/AmqpClient.cpp:110]: No I remove it

**Pull request merged**